### PR TITLE
Tester implementation

### PR DESCRIPTION
**Pull request by schoooler**

### CODE REVIEW

File: tester.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        left_bound = self.bts[quant_left]
+        right_bound = self.bts[-quant_left]
+
+        # return mean accuracy, std accuracy and 95% CI to accuracy
+        return 'metric: accuracy, mean: {:.2f}, std: {:.2f}, 95% conf interval: [{:.2f} ,{:.2f}]'.format(
+            np.mean(self.accuracy), np.std(self.accuracy), left_bound,
+            right_bound)
+
+# uncomment to test
+# if __name__ == "__main__":
```
Comment (konev-artem) [inline tester.py]: оформить как unit test

File: tester.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        # fewshotmodel is an object of class FewShotModel(*args)
+        # val_dataset is an object of class val_dataset() having method sample_episode.
+        # sample_episode returns X_train, X_test, y_train, y_test
+
+    def test(self, samples=10, bootstrap=True):
+
+        self.accuracy = []
+        for i in range(samples):
+            X_train, X_test, y_train, y_test = self.val_dataset.sample_episode()
```
Comment (konev-artem) [inline tester.py]: тут будет генератор

File: tester.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            self.model.fit(X_train, y_train)
+            self.prediction = self.model.predict(X_test)
+            self.accuracy.append(accuracy_score(y_test, self.prediction))
+
+        if bootstrap:
+            return self.bootstrap()
+
+        return 'metric: accuracy,\n mean: {:.3f},\n std: {:.3f}'.format(
```
Comment (konev-artem) [inline tester.py]: лучше возвращать не строкой, эту строку можно печатать/не печатать по параметру verbose

File: tester.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+class Tester:
+    def __init__(self, fewshotmodel, val_dataset):
+        self.model = fewshotmodel
+        self.val_dataset = val_dataset
+
+        # fewshotmodel is an object of class FewShotModel(*args)
+        # val_dataset is an object of class val_dataset() having method sample_episode.
+        # sample_episode returns X_train, X_test, y_train, y_test
+
+    def test(self, samples=10, bootstrap=True):
```
Comment (konev-artem) [inline tester.py]: переименуем bootstrap

File: tester.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+class Tester:
+    def __init__(self, fewshotmodel, val_dataset):
+        self.model = fewshotmodel
+        self.val_dataset = val_dataset
+
+        # fewshotmodel is an object of class FewShotModel(*args)
+        # val_dataset is an object of class val_dataset() having method sample_episode.
+        # sample_episode returns X_train, X_test, y_train, y_test
+
+    def test(self, samples=10, bootstrap=False, verbose = True):
```
Comment (konev-artem) [inline tester.py]: pep8 (пробелы вокруг '=')

File: tester.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            self.prediction = self.model.predict(X_test)
+            self.accuracy.append(accuracy_score(y_test, self.prediction))
+
+        if bootstrap:
+            return self.bootstrap()
+        
+        if verbose:
+            return 'metric: accuracy,\n mean: {:.3f},\n std: {:.3f}'.format(
+            np.mean(self.accuracy), np.std(self.accuracy))
+        return np.mean(self.accuracy)
```
Comment (konev-artem) [inline tester.py]: давай два числа возвращать

File: tester.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+        self.accuracy = []
+        for i in range(samples):
+            X_train, X_test, y_train, y_test = self.val_dataset.sample_episode()
+            self.model.fit(X_train, y_train)
+            self.prediction = self.model.predict(X_test)
+            self.accuracy.append(accuracy_score(y_test, self.prediction))
+
+        if bootstrap:
+            return self.bootstrap()
```
Comment (konev-artem) [inline tester.py]: тут ты все еще возвращаешь строчку (а нужно числа, как ниже)

File: tester.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import tensorflow as tf
+import numpy as np
+from sklearn.metrics import accuracy_score
+
+
+class Tester:
+    def __init__(self, fewshotmodel, val_dataset):
```
Comment (konev-artem) [inline tester.py]: режет глаз разный стиль именования: fewshotmodel без подчеркиваниий внутри имени переменной и val_dataset с подчеркиванием

File: tester.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        self.model = fewshotmodel
+        self.val_dataset = val_dataset
+
+        # fewshotmodel is an object of class FewShotModel(*args)
```
Comment (konev-artem) [inline tester.py]: я бы лучше вместо комментария делал assert(isinstance(fewshotmodel, FewShotModel)) - и проверка, и полноценная замена комментираию

File: tester.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+        if bootstrap:
+            return self.bootstrap()
+        
+        if verbose:
+            return 'metric: accuracy,\n mean: {:.3f},\n std: {:.3f}'.format(
+            np.mean(self.accuracy), np.std(self.accuracy))
+        return np.mean(self.accuracy)
+
+        # calculate 95% CI using bootstrap
```
Comment (konev-artem) [inline tester.py]: форматирование съехало

File: tester.py [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+    def bootstrap(self, sz=1000):
```
Comment (konev-artem) [inline tester.py]: не понимаю, почему ты берешь sz как параметр функции, а accuracy как член класса
я бы и то, и то передавал параметром
